### PR TITLE
Call explicitly fclose in copy_file

### LINKED ISSUE #1535: copy does not preserve mtime anymore / missing fclose
**Setup (please complete the following information):**
 - Distribution: Gentoo
 - Distribution release: N/A
 - Geeqie version [geeqie --version]: 2.4 and 2.5

**Describe the bug**
When an image is copied to another directory with the copy window (ctrl-c), mtime of the created file is set to the current time

**To reproduce**
Select an image, type ctrl-c, choose an output directory, check mtime of thee created file in the terminal with `ls -l` or `stat`

**Expected behavior**
mtime of the created file should be the same than mtime of the original file, as it was doing in v2.1 (and as the code is specifying with the last `TRUE` of `copy_file_attributes(s, t, TRUE, TRUE);` in `copy_file` in `src/ui-fileops.cc`)

**Screenshots**
N/A

**Image sources**
N/A

**Error logs or seg. fault files**
N/A

**Additional context**

Since I installed v2.4 (after v2.1) the mtime of copied images (with ctrl-c) is not preserved anymore, but reset to the current time. After analyzing the code I didn't find any obvious issue (i.e. checked that it was supposed to be done), so I ended up running strace and obtained that:

```
...
write(17, "exl#\205\367"..., 12288) = 12288
read(16, "}\257\333\332"..., 16384) = 15071
read(16, "", 4096)                      = 0
write(17, "}\257\333\332"..., 4096) = 4096
write(17, "\324\255+&\306"..., 8192) = 8192
read(16, "", 16384)                     = 0
rename("/tmp/test_img/20241021_185345.jpg.tmp_5CZIW2", "/tmp/test_img/20241021_185345.jpg") = 0
newfstatat(AT_FDCWD, "/tmp/20241021_185345.jpg", {st_mode=S_IFREG|0660, st_size=6044383, ...}, 0) = 0
chown("/tmp/test_img/20241021_185345.jpg", 1000, 1000) = 0
chmod("/tmp/test_img/20241021_185345.jpg", 0100660) = 0
utimensat(AT_FDCWD, "/tmp/test_img/20241021_185345.jpg", [{tv_sec=1730633761, tv_nsec=0} /* 2024-11-03T12:36:01+0100 */, {tv_sec=1730795817, tv_nsec=0} /* 2024-11-03T09:36:57+0100 */], 0) = 0
write(17, "\330\\\26\202\344\\+\346Q#\204\371\f\"H\324\234p\244\201\315Os\247\t\354\256mbd_\263"..., 2783) = 2783
close(17)                               = 0
close(16)                               = 0
```

The utime call is correctly issued, with the right parameters, but it is issued before the file is closed, and even before all the data is written to the file. I assumed that it could be the origin of my problem, as the file is modified after and resetting mtime.

I went back to the code, and noticed that the files were not explicitly closed, and by looking at the history I found that in b8f365d5e7be4b7c861 (right after v2.1), the existing explicit `fclose` are removed and replaced by setting `fclose_safe` as the deleter of the `unique_ptr`, which cause the file descriptors to be closed at the end of `copy_file` *after* the call to `copy_file_attributes`.

However it wasn't the same version than in master, and I found that later in 428bd7b1ed69754e (before v2.5), the `unique_ptr` are removed, but nothing is added back to close the file descriptors (it seems to be managed by `glib_autoptr_cleanup`), which explains that the issue is still present in v2.5, as I tested afterwards.

So I tried to explicitly close the file descriptors in v2.5 (I'll open a PR):

```
--- a/src/ui-fileops.cc
+++ b/src/ui-fileops.cc
@@ -583,6 +583,9 @@ gboolean copy_file(const gchar *s, const gchar *t)
                        }
                }
 
+       fclose(fi); fi = nullptr;
+       fclose(fo); fo = nullptr;
+
        if (rename(randname, tl) < 0)
                {
                unlink(randname);
```

And it works, i.e. mtime is preserved, and `strace` shows that now everything is written and closed before calling `utime` (and before renaming, which didn't seem to be an issue, but still make more sense):

```
write(21, "exl#\205\367"..., 12288) = 12288
read(20, "}\257\333\332"..., 16384) = 15071
read(20, "", 4096)                      = 0
write(21, "}\257\333\332"..., 4096) = 4096
write(21, "\324\255+&\306"..., 8192) = 8192
read(20, "", 16384)                     = 0
close(20)                               = 0
write(21, "\330\\\26\202\344\\+\346Q#\204\371\f\"H\324\234p\244\201\315Os\247\t\354\256mbd_\263"..., 2783) = 2783
close(21)                               = 0
rename("/tmp/test_img/20241021_185345.jpg.tmp_BV4XW2", "/tmp/test_img/20241021_185345.jpg") = 0
newfstatat(AT_FDCWD, "/tmp/20241021_185345.jpg", {st_mode=S_IFREG|0660, st_size=6044383, ...}, 0) = 0
chown("/tmp/test_img/20241021_185345.jpg", 1000, 1000) = 0
chmod("/tmp/test_img/20241021_185345.jpg", 0100660) = 0
utimensat(AT_FDCWD, "/tmp/test_img/20241021_185345.jpg", [{tv_sec=1730830949, tv_nsec=0} /* 2024-11-05T19:22:29+0100 */, {tv_sec=1730795817, tv_nsec=0} /* 2024-11-05T09:36:57+0100 */], 0) = 0
```

### PR DESCRIPTION
**Pull request by cyril42e**
Fixes #1535, details in the issue. Just restored the code as it was before b8f365d5e7be4b. Tested on v2.5 tag.

### CODE REVIEW

File: src/ui-fileops.cc [hunk 1]
```diff
@@ -583,6 +583,9 @@ gboolean copy_file(const gchar *s, const gchar *t)
 			}
 		}
 
+	fclose(fi); fi = nullptr;
```
Comment (qarkai) [inline src/ui-fileops.cc:586]: Could you please add comment why we explicitly close files here?

Comment (cyril42e) [reply src/ui-fileops.cc:589]: Good point, let me know if the level of details is suitable.

Comment (qarkai) [reply src/ui-fileops.cc:589]: LGTM

**Comment by caclark** [comment]
Implemented in commit 47d3a069786a52a0dbe89e15e30ba2f0d03be854

**Pull request closed without merging**